railgun/test/integration/knative_test.go

```diff
@@ -165,6 +167,26 @@ func configKnativeDomain(ctx context.Context, proxy string, cluster kind.Cluster
 }
 
 func accessKnativeSrv(ctx context.Context, proxy string, t *testing.T) bool {
+	knativeCli, err := knativenetworkingversioned.NewForConfig(cluster.Config())
+	if err != nil {
+		return false
+	}
+	ingCli := knativeCli.NetworkingV1alpha1().Ingresses("default")
+	assert.Eventually(t, func() bool {
+		curIng, err := ingCli.Get(ctx, "helloworld-go", metav1.GetOptions{})
+		if err != nil || curIng == nil {
+			return false
+		}
+		conds := curIng.Status.Status.GetConditions()
+		for _, cond := range conds {
+			if cond.Type == apis.ConditionReady && cond.Status == v1.ConditionTrue {
+				t.Logf("knative ingress status is ready.")
+				return true
+			}
+		}
+		return false
+	}, 120*time.Second, 1*time.Second, true)
```

Comment:
Q: how was 2 minutes here derived, is that the result of some testing or is it semi-arbitrary?

Reply:
The estimated worst case based on manual testing on local kind environment.